File: src/main/java/com/github/kayjamlang/core/expressions/ReturnExpression.java
package com.github.kayjamlang.core.expressions;

import com.github.kayjamlang.core.opcodes.AccessType;

public class ReturnExpression extends Expression {

    /**
     * Return value
     * Can be null, if return void value
     */
    public final Expression expression;

    public ReturnExpression(Expression expression, int line) {
        super(AccessType.NONE, line);
        this.expression = expression;
    }

    @Override
    public String toString() {
        return "Return{" +
                "expression=" + expression +
                ", identifier=" + accessType +
                ", line=" + line +
                '}';
    }
}

File: src/main/scala/com/github/kayjamlang/core/expressions/ReturnExpression.scala
package com.github.kayjamlang.core.expressions

import com.github.kayjamlang.core.opcodes.AccessType

class ReturnExpression(val expression: Expression, line: Int) extends Expression(AccessType NONE, line) {
  override def toString: String = s"Return{expression=$expression, identifier=$accessType, line=$line}"
}